### ISSUE TITLE
RecordDetail - Enhancement - Ability to pass a Parent or Lookup Record Id when creating a new record

### ISSUE THREAD

**Issue by SDJCorp** [alexed1/LightningFlowComponents#491]
Was trying to use the Record Detail Component to add a new child object to an Opportunity but when I add it, the Opportunity field isn't populated.

The work around was to create the record first, with the required fields defaulted, and the opportunity ID filled in, then call the Record Detail Component with the newly created child record ID for the user to complete, but that could lead to issues when the user cancels out of the screen... so... 

It would be much easier to just pass in the API name and record ID of the parent/lookup field that you want to use.

Also, for junction objects, maybe have 2 sets of field names and IDs?

**Comment by JodieM** [comment]
Yes, I would love this. Plus the ability to set default values. Then it would be _amost_ perfect.

**Comment by adwhite75** [comment]
@SDJCorp / @JodieM   could you provide some screenshots/mockups of how you'd expect this might work/look?

**Comment by JodieM** [comment]
I have a child record of Account, named Development (Development__c)

I want to use this excellent component to do a "Quick Create"... eg I don't want to see ALL Fields on the Development object because the Development (project) runs for many years and there is a lot of data to capture over the years. When the record is first created there are just a few key fields that need to be created. 

I create this record from another record (eg Case - not real, but let's go with it). Case has a lookup to Account already. 

So in my Flow I pass the Record ID of the Case into the Flow, then Get Records on Case, to return the AccountID and store that in a variable {!varAccountId}. 

I then want to show a screen showing this Record Detail Component to show just the fields required to be entered at this stage, but prefilling the Account Name. 

So when setting up the Component I need to enter the variable of the related record to pass in the ID, and I need to know which field that variable goes to. 

See mockup image.
![Mockup](https://user-images.githubusercontent.com/2966583/103432999-9a471d80-4c3d-11eb-8e86-691543d0b311.png)

However, this could even be much smarter... look at [Dan Howell's Navigate Button](https://github.com/danhowellnz/NavigateButton) and how that does the Create New Record link... that works really well. That would allow multiple fields to be related. Dan's approach could be used instead of the simple Record Field List entry and allow all values to be pre-filled. (Eg my Development has lookups to 3 Accounts and 4 Contacts for various reasons).  

Here is how Navigate Button is set up. 
New Record Settings:
![NewRecord](https://user-images.githubusercontent.com/2966583/103433102-53f2be00-4c3f-11eb-9114-66c3c5f57727.png)

TextTemplate with name, value pairs
![TextTemplate](https://user-images.githubusercontent.com/2966583/103433114-65d46100-4c3f-11eb-8e26-f53e48222bfe.png)

I can do this exact scenario with Navigate Button and Navigate Button is good in that it shows the real record in Edit mode, but as I noted I would like to have a "Quick Create" style of thing where only the necessary fields are added. 

Note also that this could potentially be accomplished with Dynamic Forms, but even in Spring '21 Dynamic Forms are not ready for prime time and are also not available on Standard Objects. Your component works with Standard Objects too.
